programs/mango-v4/src/state/perp_market.rs

```diff
@@ -256,13 +256,13 @@ impl PerpMarket {
         Ok(price)
     }
 
-    pub fn oracle_price_and_slot(
+    pub fn oracle_price_and_meta(
         &self,
         oracle_acc: &impl KeyedAccountReader,
         staleness_slot: Option<u64>,
-    ) -> Result<(I80F48, u64)> {
+    ) -> Result<(I80F48, u64, I80F48, oracle::OracleType)> {
```

Comment:
suggestion: make a struct OracleState (or something) with price, slot, confidence, type so this function can have a clearer return va lue

Reply:
Thanks for that suggestion - I have implemented it.